mpire/pool.py

```diff
@@ -233,10 +265,17 @@ def _restart_workers(self) -> None:
 
             # Check if it needs restarting
             if restart_worker:
+                # Obtain results from exit results queue (should be done before joining the worker)
+                if self._exit_results_queues:
+                    with DisableKeyboardInterruptSignal():
```

Comment:
Oh whoops, I meant to delay it, not disable it. I'll fix that.

Multiprocessing queues don't work nicely with keyboard interrupts. Once I fixed it it will raise a keyboardinterrupt once the block finishes